### PR TITLE
modal added and refined existing features

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
# React+Redux Blog App Frontend:
Description: Modal has been added. Click on `Add Blog` to see the modal and click close to go back.
Also, User is only able to edit those blogs which are posted by him.

Installation Instructions:

1. change directory to `frontend` 
2. run `npm install`
3. run npm start

Screenshot from 2019-09-20 17-20-11

### CODE REVIEW

File: frontend/.gitignore [hunk 1]
```diff
@@ -16,3 +16,4 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 
+**.idea/**
```
Comment (etechdevai) [inline frontend/.gitignore:19]: add empty line at end of file

File: frontend/src/components/modal/CreateBlogModal.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import './Modal.css';
+
+const CreateBlogModal = ({handleClose, show, children}) => {
```
Comment (etechdevai) [inline frontend/src/components/modal/CreateBlogModal.js:4]: why class component is not used ?

File: frontend/src/components/blog/UpdateForm.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ import React, {Component} from "react";
 import {connect} from "react-redux";
 import PropTypes from "prop-types";
 import {updateBlog} from "../../actions/blogList";
+import {Redirect} from "react-router-dom";
 
 class UpdateForm extends Component {
```
Comment (etechdevai) [inline frontend/src/components/blog/UpdateForm.js:7]: Use proper descriptive class names

File: frontend/src/components/blog/Dashboard.js [hunk 4]
```diff
@@ -1,13 +1,37 @@
-import React, {Fragment} from "react";
-import CreateForm from "./CreateForm";
+import React, {Component, Fragment} from "react";
 import BlogList from "./BlogList";
+import CreateForm from "./CreateForm";
+import CreateBlogModal from '../modal/CreateBlogModal'
```
Comment (etechdevai) [inline frontend/src/components/blog/Dashboard.js:4]: Why is there Modal at end of files/classes ?

File: frontend/src/components/blog/BlogList.js [hunk 5]
```diff
@@ -9,8 +9,7 @@ class BlogList extends Component {
 
     static propTypes = {
         blogList: PropTypes.array.isRequired,
-        getBlogList: PropTypes.func.isRequired,
-        deleteBlog: PropTypes.func.isRequired
+        getBlogList: PropTypes.func.isRequired
```
Comment (etechdevai) [inline frontend/src/components/blog/BlogList.js:12]: what is the reason of using pro-types here?

File: frontend/src/App.js [hunk 6]
```diff
@@ -13,13 +13,6 @@ import {loadUser} from "./actions/auth";
 import BlogItem from "./components/blog/BlogItem";
 
 class App extends Component {
-
-    state = {
-        blog_list: [],
```
Comment (etechdevai) [inline frontend/src/App.js:18]: Where is this functionality moved ? and as per last review you were required to move these numeric values to constants?

**Review by etechdevai — CHANGES_REQUESTED** [review]

**Comment by etechdevai** [comment]
@Ali-D-Akbar add proper name for PR describing it's functionality. and what is reason of adding modal on frontend side ?

File: frontend/src/actions/blogList.js [hunk 7]
```diff
@@ -23,6 +25,7 @@ export const getBlogList = () => (dispatch, getState) => {
 //GET_BLOGITEM
 
 export const getBlogItem = (id) => (dispatch, getState) => {
+    console.log(id);
```
Comment (zubair-arbi) [inline frontend/src/actions/blogList.js:28]: nit: remove this line

File: frontend/src/components/modal/CreateBlogModal.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    const showHideClassName = show ? "card modal display-block" : "modal display-none";
+
+    return (
+        <div className={showHideClassName}>
```
Comment (zubair-arbi) [inline frontend/src/components/modal/CreateBlogModal.js:8]: Lets update the layout of the form like bootstrap:
Screen Shot 2019-09-20 at 6 04 23 PM

**Review by zubair-arbi — CHANGES_REQUESTED** [review]

File: frontend/src/actions/blogList.js (hunk 7 above)
Comment (Ali-D-Akbar) [reply frontend/src/actions/blogList.js:28]: done.

File: frontend/src/components/modal/CreateBlogModal.js (hunk 8 above)
Comment (Ali-D-Akbar) [reply frontend/src/components/modal/CreateBlogModal.js:8]: done

**Comment by zubair-arbi** [comment]
@Ali-D-Akbar why you merged this PR, when you didn't got 2 thumbs up?

The rule is:

1. make changes
2. get review
3. address review
4. repeat from step 2 until all reviewer's feedback is addressed
5. get 2 thumbs up
6. merge the changes
7. delete merged PR

**Pull request merged**